src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,206 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { useLocation, useSearchParams, NavLink } from 'react-router-dom';
+import debounce from 'lodash/debounce';
+
+import PeopleTable from './PeopleTable';
+import NewPerson from './NewPerson';
+
+import { getPeople } from '../api/api';
+
+import PersonEnum from '../enums/PersonEnum';
+import getSearchWith from '../utils/getSearchWith';
+
+const linkPersonWithParents = (person: Person, peopleArray: Person[]) => {
+  const mother = peopleArray.find(possibleParent => (
+    possibleParent.name === person.motherName
+  ));
+  const father = peopleArray.find(possibleParent => (
+    possibleParent.name === person.fatherName
+  ));
+
+  return {
+    ...person,
+    mother: mother || null,
+    father: father || null,
+  };
+};
+
+const loadData = (): Promise<Person[]> => {
+  return getPeople()
+    .then(response => {
+      return response.map((person, _, arr) => (
+        linkPersonWithParents(person, arr)
+      ));
+    });
+};
+
+const PeoplePage: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const isFormOpen = useLocation().pathname === '/people/new';
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isError, setIsError] = useState<boolean>(false);
+
+  const appliedQuery = searchParams.get('query') || '';
+  const sortBy = searchParams.get('sortBy') || '';
+  const sortOrder = searchParams.get('sortOrder') || '';
+
+  const [query, setQuery] = useState(appliedQuery);
+
+  useEffect(() => {
+    loadData()
+      .then(loadedPeople => setPeople(loadedPeople))
+      .catch(() => setIsError(true));
+  }, []);
+
+  const applyQuery = useCallback(
+    debounce((newQuery: string) => {
+      setSearchParams(getSearchWith({ query: newQuery }, searchParams));
+    }, 500),
+    [],
+  );
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handleNewPersonSubmit = (newPerson: Person) => {
+    setPeople(prevPeople => {
+      return ([
+        ...prevPeople,
+        linkPersonWithParents(newPerson, prevPeople),
+      ]);
+    });
+  };
+
+  const preparePeople = () => {
+    const preparedPeople = people.filter(person => (
+      person.name.toLowerCase()
+        .includes(appliedQuery.toLowerCase())
```

Comment:
you use 3 times `appliedQuery.toLowerCase()`, let's create a variable for this expression